### PR TITLE
Clean up page controllers

### PR DESCRIPTION
**Pull request by aschempp**
Partially duplicates https://github.com/contao/contao/pull/4291, so either needs rebase after merging or remove the respective portions

### CODE REVIEW

**Review by Toflar — DISMISSED** [review]

**Review by fritzmg — DISMISSED** [review]

**Review by fritzmg — DISMISSED** [review]

File: core-bundle/src/Resources/contao/config/config.php [hunk 1]
```diff
@@ -386,7 +385,6 @@
 	'regular'   => PageRegular::class,
 	'forward'   => PageForward::class,
 	'redirect'  => PageRedirect::class,
-	'root'      => PageRoot::class,
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/config/config.php:389]: How can I add root pages in the site structure if you remove this?

File: core-bundle/src/Resources/contao/controllers/FrontendIndex.php [hunk 2]
```diff
@@ -117,27 +62,6 @@ public function renderPage(PageModel $pageModel)
...
-		}
-
-		// Check whether there are domain name restrictions
-		if ($objPage->domain && $objPage->domain != Environment::get('host'))
-		{
-			System::getContainer()->get('monolog.logger.contao.error')->error('Page ID "' . $objPage->id . '" was requested via "' . Environment::get('host') . '" but can only be accessed via "' . $objPage->domain . '" (' . Environment::get('base') . Environment::get('request') . ')');
-
-			throw new PageNotFoundException('Page not found: ' . Environment::get('uri'));
-		}
-
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/controllers/FrontendIndex.php:119]: Are you 100% sure that all this (and the lines above) can be removed? All the checks look valid to me.

File: core-bundle/src/Resources/contao/config/config.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Resources/contao/config/config.php:389]: https://github.com/contao/contao/blob/5.x/core-bundle/src/Controller/Page/RootPageController.php

File: core-bundle/src/Resources/contao/controllers/FrontendIndex.php (hunk 2 above)
Comment (aschempp) [reply core-bundle/src/Resources/contao/controllers/FrontendIndex.php:119]: they are all done by the routing component. A page is never matched if it is not published, the prefix does not match or the domain is different.

**Comment by leofeyer** [comment]
Can you rebase the pull and resolve the conflicts, please?

**Review by Toflar — APPROVED** [review]

**Pull request merged**